addons/website/models/ir_module_module.py

```diff
@@ -76,7 +76,7 @@ def write(self, vals):
 
                     -> We want to upgrade every website using this theme.
         """
-        if request and request.db and request.context.get('apply_new_theme'):
+        if request and request.db and request.env and request.context.get('apply_new_theme'):
```

Comment:
You have to justify that we can skip `with_context` instead of raising some error